File: ExampleApp/app/src/main/java/com/motiv/example/PostsFragmentContract.java
package com.motiv.example;

import androidx.fragment.app.*;
import dagger.*;
import dagger.android.*;
import dagger.android.support.*;
import javax.inject.*;

public interface PostsFragmentContract {

    public interface View {

        public void postsAdaptersetData(java.util.List<com.motiv.example.Post> arg0);

        public void navigationControllerstartPostDetailsActivity(java.lang.String arg0);
    }

    public interface Presenter {

        public void goApigetPostsList();

        public void eloonItemClick(int argument0, com.motiv.example.Post argument1);
    }
}

File: ExampleApp/app/src/main/java/com/motiv/example/PostsFragmentContract.kt
 
package com.motiv.example
import kotlin.collections.List

public interface PostsFragmentContract {

    public interface View {

        public fun postsAdaptersetData(arg0: List<com.motiv.example.Post>): Unit
        public fun navigationControllerstartPostDetailsActivity(arg0: String): Unit
    }

    public interface Presenter {

        public fun goApigetPostsList(): Unit
        public fun eloonItemClick(argument0: Int, argument1: com.motiv.example.Post): Unit
    }
}
